tests/conftest.py

```diff
@@ -1,196 +1,33 @@
-    parser.addoption("--sku", action="store", default=None, help="SKU mode")
+    parser.addoption("--setup", action="store", default=None, help="Setup description (Path to the json file).")
 
 
 @pytest.fixture(scope="session")
 def exec_params(request):
     config_param = {}
-    config_param["server"] = request.config.getoption("--sai-server")
+    config_param["setup"] = init_setup(request.config)
     config_param["traffic"] = request.config.getoption("--traffic")
-    config_param["saivs"] = request.config.getoption("--saivs")
     config_param["loglevel"] = request.config.getoption("--loglevel")
-    config_param["asic"] = request.config.getoption("--asic")
-    config_param["target"] = request.config.getoption("--target")
-    config_param["sku"] = request.config.getoption("--sku")
+    logging.getLogger().setLevel(getattr(logging, config_param["loglevel"].upper(), "INFO"))
     return config_param
 
 
-def logging_setup(config):
-    """
-    Set up logging based on config
-    """
-
-    logging.getLogger().setLevel(DEBUG_LEVELS[config["debug"]])
-
-    if config["log_dir"] != None:
-        if os.path.exists(config["log_dir"]):
-            import shutil
-            shutil.rmtree(config["log_dir"])
-        os.makedirs(config["log_dir"])
-    else:
-        if os.path.exists(config["log_file"]):
-            os.remove(config["log_file"])
-
-    ptf.open_logfile('main')
-
-
 @pytest.fixture(scope="session")
 def npu(exec_params):
-    npu = None
-    exec_params["asic_dir"] = None
-
-    if exec_params["asic"] == "generic":
-        npu = SaiNpu(exec_params)
-    else:
-        asic_dir = None
-        npu_mod = None
-        module_name = "sai_npu"
-
-        try:
-            asic_dir = glob.glob("../npu/**/" + exec_params["asic"] + "/", recursive=True)
-            asic_dir = asic_dir[0]
-            exec_params["asic_dir"] = asic_dir
-        except:
-            logging.critical("Failed to find " + exec_params["asic"] + " NPU folder")
-            sys.exit(1)
-
-        try:
-            npu_mod = imp.load_module(module_name, *imp.find_module(module_name, [asic_dir]))
-        except:
-            logging.info("No {} specific 'sai_npu' module defined..".format(exec_params["asic"]))
-            try:
-                npu_mod = imp.load_module(module_name, *imp.find_module(module_name, [asic_dir + "../"]))
-            except:
-                logging.warn("No NPU specific 'sai_npu' module defined.")
-
-        if npu_mod is not None:
-            try:
-                npu = npu_mod.SaiNpuImpl(exec_params)
-            except:
-                logging.critical("Failed to instantiate 'sai_npu' module for {}".format(exec_params["asic"]))
-                sys.exit(1)
-        else:
-            logging.info("Falling back to the default 'sai_npu' module..")
-            npu = SaiNpu(exec_params)
-
+    npu = exec_params["setup"]["NPU"][0]
```

Comment:
Preferable not to use global configuration to pass objects globally

Reply:
Please explain

Reply:
exec_params is loaded configuration and NPU is part of some topology model. Configurations have to be immutable not to confuse their users. Models usually carry some state. Configurations are used on initialization and models are used during the application run.